specification/scripts/simplify_specification.dart

```diff
@@ -0,0 +1,442 @@
+      if (line.startsWith(r"\end{dartCode}")) {
+        inDartCode = false;
+      }
+    }
+  }
+
+  /// Remove non-normative text elements.
+  /// This method removes `\commentary{...}` and `\rationale{...}`,
+  /// and it removes `\BlindDefineSymbol{...}`. It does not make
+  /// any attempts to balance brace begin and brace end characters,
+  /// it trusts the indentation to reflect this structure correctly.
+  void removeNonNormative() {
+    for (int i = 0; i < length; ++i) {
+      final line = this[i];
+      if (line == null) continue;
+      if (line.startsWith(r"\BlindDefineSymbol{")) {
+        this[i] = null;
+        continue;
+      }
+      final match = _commentaryRationaleRegExp.firstMatch(line);
+      if (match != null) {
+        final matchParenthesizedOneliner = _parenBracesRexExp.firstMatch(line);
+        if (matchParenthesizedOneliner != null) {
+          if (matchParenthesizedOneliner.start == 0 &&
+              matchParenthesizedOneliner.end == line.length) {
+            this[i] = null;
+          } else {
+            this[i] = line.replaceRange(
+              matchParenthesizedOneliner.start,
+              matchParenthesizedOneliner.end,
+              '',
+            );
+          }
+        } else {
+          final matchOneliner = _bracesRegExp.firstMatch(line);
+          if (matchOneliner != null) {
+            if (matchOneliner.start == 0 && matchOneliner.end == line.length) {
+              this[i] = null;
+            } else {
+              this[i] = line.replaceRange(
+                matchOneliner.start,
+                matchOneliner.end,
+                '',
+              );
+            }
+          } else {
+            final lineStart = _lineStart(line);
+            while (i < length && this[i]?.startsWith(lineStart) != true) {
+              this[i] = null;
+              ++i;
+            }
+            if (i < length) this[i] = null;
+          }
+        }
+      }
+    }
+  }
+
+  void removeSpuriousEmptyLines() {
+    final length = this.length;
+    bool previousLineWasEmpty = false;
+    for (int i = 0; i < length; ++i) {
+      var line = this[i];
+      if (line == null) continue;
+      if (line.isEmpty) {
+        if (previousLineWasEmpty) {
+          this[i] = null;
+        } else {
+          previousLineWasEmpty = true;
+        }
+      } else {
+        previousLineWasEmpty = false;
+      }
+    }
+  }
+
+  void removeTrailingWhitespace() {
+    for (int i = 0; i < length; ++i) {
+      final line = this[i];
+      if (line == null) continue;
+      if (line.isNotEmpty && _isWhitespace(line, line.length - 1)) {
+        this[i] = line.trimRight();
+      }
+    }
+  }
+
+  /// Return the index of the first line, starting with [startIndex],
+  /// that contains the command `\item`. Note that this implies
+  /// `this[i] != null` where `i` is the returned value.
+  ///
+  /// This method does not attempt to balance `\begin{}`/`\end{}`
+  /// pairs,
```

Comment:
Could `,` be a typo here, and a `.` was intended?

Reply:
Indeed.